src/contentmanager.cpp

```diff
@@ -189,18 +193,51 @@ QString ContentManager::downloadBook(const QString &id)
     return QString::fromStdString(download->getDid());
 }
 
-void ContentManager::eraseBook(const QString& id)
+void ContentManager::eraseBookFilesFromComputer(const QString fileSelection)
 {
-    kiwix::Book book = mp_library->getBookById(id);
     QString dirName = QString::fromUtf8(getDataDirectory().c_str());
-    QString fileSelection = QString::fromUtf8(getLastPathElement(book.getPath()).c_str()) + "*";
     QDir dir(dirName, fileSelection);
     for(const QString& filename: dir.entryList()) {
         dir.remove(filename);
     }
+}
+
+void ContentManager::eraseBook(const QString& id)
+{
+    kiwix::Book book = mp_library->getBookById(id);
+    QString fileSelection = QString::fromUtf8(getLastPathElement(book.getPath()).c_str()) + "*";
+    eraseBookFilesFromComputer(fileSelection);
+    mp_library->removeBookFromLibraryById(id);
+    mp_library->save();
+    emit(mp_library->booksChanged());
+}
+
+void ContentManager::pauseBook(const QString& id)
+{
+    auto& b = mp_library->getBookById(id);
+    auto download = mp_downloader->getDownload(b.getDownloadId());
+    if (download->getStatus() == kiwix::Download::K_ACTIVE)
+        download->pauseDownload();
+}
+
+void ContentManager::resumeBook(const QString& id)
+{
+    auto& b = mp_library->getBookById(id);
+    auto download = mp_downloader->getDownload(b.getDownloadId());
+    if (download->getStatus() == kiwix::Download::K_PAUSED)
+        download->resumeDownload();
+}
+
+void ContentManager::cancelBook(const QString& id)
+{
+    auto& b = mp_library->getBookById(id);
+    auto download = mp_downloader->getDownload(b.getDownloadId());
+    QString fileSelection = QString::fromUtf8(getLastPathElement(download->getPath()).c_str()) + "*";
+    eraseBookFilesFromComputer(fileSelection);
     mp_library->removeBookFromLibraryById(id);
     mp_library->save();
     emit(mp_library->booksChanged());
+    download->cancelDownload();
```

Comment:
The cancel should probably be called before we remote the file from the fs or update the view.
Else, we could have a race condition where aria2 try to update the file on the fs while we are deleting it.